cli/command/cli.go

```diff
@@ -419,8 +419,8 @@ type ClientInfo struct {
 }
 
 // NewDockerCli returns a DockerCli instance with IO output and error streams set by in, out and err.
-func NewDockerCli(in io.ReadCloser, out, err io.Writer, isTrusted bool, containerizedFn func(string) (clitypes.ContainerizedClient, error)) *DockerCli {
-	return &DockerCli{in: NewInStream(in), out: NewOutStream(out), err: err, contentTrust: isTrusted, newContainerizeClient: containerizedFn}
+func NewDockerCli(in io.ReadCloser, out, err io.Writer, containerizedFn func(string) (clitypes.ContainerizedClient, error)) *DockerCli {
```

Comment:
If we're changing this Exported function anyway, would it  make sense to rewrite it to use functional arguments?

Things like this start to get ugly;

```go
stdin, stdout, stderr := term.StdStreams()
dockerCli := command.NewDockerCli(stdin, stdout, stderr, nil)
```

And with functional arguments, it could look like, e.g. (we may want error-handling in there, but  just as a quick thought);

```go
func NewDockerCli(ops ...func(*DockerCli)) *DockerCli
```
```go
dockerCli := command.NewDockerCli(
	WithStandardStreams,
	WithContentTrust,
	WithContainerizedClient,
)
```

(this could make the interface more stable, as new options can be added without having to change the interface)

Reply:
Just thought about this concrete example: `duffle` the CNAB client which uses the docker cli to pull images. See https://github.com/deislabs/duffle/blob/master/pkg/driver/docker_driver.go#L100
And I don't think `duffle` will ever be a docker cli plugin 😄

Reply:
AFAICS (and I only had a quick look) they only want a cli so they can call `cli.Client().ImagePull()` (and other similar constructs). They'd be as well to go at the `github.com/docker/docker/client` interfaces (which is effectively what `cli.Client()` gives you), the `cli` isn't really adding very much of use here.

And referring back to https://github.com/docker/cli/pull/1564/#discussion_r248236365 (which is what I tried to link to in my comment 2 ago but linked just to the PR by mistake), if we consider such use of the `NewDockerCli` to be fine, I still don't think it makes sense for other tools to just hardcode content trust to off.